### PR TITLE
[4.0] Enable button when returning from article

### PR DESCRIPTION
**Pull request by maliouris**
Pull Request for Issue # .

### Summary of Changes

Fix disabled buttons to enabled when an element is already selected on load.

### Testing Instructions

Steps to reproduce the issue. Go to articles, select an article and then press the name of it to edit it. Press the back button to go back and you will see that the buttons change status and batch are disabled and cant be enabled even if you deselect it and then you select it again.

### Expected result

checked-works

### Actual result

checked-doesnt-work

### CODE REVIEW

File: build/media_source/system/js/joomla-toolbar-button.w-c.es6.js [hunk 1]
```diff
@@ -44,15 +44,16 @@ window.customElements.define('joomla-toolbar-button', class extends HTMLElement
...
-    if (this.listSelection) {
-      this.setDisabled(true);
-    }
-
     if (this.listSelection) {
       if (!this.formElement) {
         throw new Error(`The form "${formSelector}" is required to perform the task, but the form was not found on the page.`);
       }
 
+      //If nothing is selected disable the list
```
Comment (Quy) [inline build/media_source/system/js/joomla-toolbar-button.w-c.es6.js]:
```suggestion
      // If nothing is selected disable the list
```

File: build/media_source/system/js/joomla-toolbar-button.w-c.es6.js [hunk 2]
```diff
@@ -44,15 +44,16 @@ window.customElements.define('joomla-toolbar-button', class extends HTMLElement
...
+      if(this.formElement.boxchecked.value!=1){
```
Comment (Quy) [inline build/media_source/system/js/joomla-toolbar-button.w-c.es6.js]:
```suggestion
      if (this.formElement.boxchecked.value != 1) {
```

File: build/media_source/system/js/multiselect.es6.js [hunk 3]
```diff
@@ -26,8 +26,17 @@
         }
 
         if (this.rows.length) {
+          var i=0;
```
Comment (Quy) [inline build/media_source/system/js/multiselect.es6.js]:
```suggestion
          var i = 0;
```

File: build/media_source/system/js/multiselect.es6.js [hunk 4]
```diff
@@ -26,8 +26,17 @@
...
           this.rows.forEach((row) => {
+            var currentCheckBox = this.checkallToggle ? i + 1 : i;
+            var isChecked = this.boxes[currentCheckBox].checked;
+            //If something is checked on load, update the form
```
Comment (Quy) [inline build/media_source/system/js/multiselect.es6.js]:
```suggestion
            // If something is checked on load, update the form
```

File: build/media_source/system/js/multiselect.es6.js [hunk 5]
```diff
@@ -26,8 +26,17 @@
...
+            if(isChecked){
```
Comment (Quy) [inline build/media_source/system/js/multiselect.es6.js]:
```suggestion
            if (isChecked) {
```

**Comment by infograf768** [comment]
Please use spaces instead of tabs...

File: build/media_source/system/js/joomla-toolbar-button.w-c.es6.js [hunk 6]
```diff
@@ -44,15 +44,16 @@ window.customElements.define('joomla-toolbar-button', class extends HTMLElement
...
-      this.setDisabled(true);
-    }
-
     if (this.listSelection) {
       if (!this.formElement) {
         throw new Error(`The form "${formSelector}" is required to perform the task, but the form was not found on the page.`);
       }
 
+      // If nothing is selected disable the list
+      if(this.formElement.boxchecked.value < 1){
```
Comment (infograf768) [inline build/media_source/system/js/joomla-toolbar-button.w-c.es6.js]: space missing between if and parenthesis
space missing before brace

File: build/media_source/system/js/multiselect.es6.js [hunk 7]
```diff
@@ -26,8 +26,20 @@
...
+            var currentCheckBox = this.checkallToggle ? i + 1 : i;
+	    var checkbox = this.boxes[currentCheckBox];
+	    if (checkbox) {
+                var isChecked = checkbox.checked;
+                // If something is checked on load, update the form
+            	if (isChecked) {
+                    Joomla.isChecked(true);
+              	    this.changeBg(row,true);
+            	}
+						}
```
Comment (infograf768) [inline build/media_source/system/js/multiselect.es6.js]: This whole part above:
change tabs to spaces.
var should be const

```
drone/src/github.com/joomla/joomla-cms/build/media_source/system/js/multiselect.es6.js
  29:11  error  All 'var' declarations must be at the top of the function scope  vars-on-top
  29:11  error  Unexpected var, use let or const instead                         no-var
  31:13  error  Unexpected var, use let or const instead                         no-var
  32:1   error  Unexpected tab character                                         no-tabs
  32:2   error  Mixed spaces and tabs                                            no-mixed-spaces-and-tabs
  32:6   error  Unexpected var, use let or const instead                         no-var
  33:1   error  Unexpected tab character                                         no-tabs
  33:2   error  Mixed spaces and tabs                                            no-mixed-spaces-and-tabs
  34:1   error  Expected indentation of 14 spaces but found 16                   indent
  34:17  error  Unexpected var, use let or const instead                         no-var
  34:17  error  All 'var' declarations must be at the top of the function scope  vars-on-top
  35:1   error  Expected indentation of 14 spaces but found 16                   indent
  36:2   error  Mixed spaces and tabs                                            no-mixed-spaces-and-tabs
  36:13  error  Unexpected tab character                                         no-tabs
  37:1   error  Expected indentation of 16 spaces but found 20                   indent
  38:2   error  Mixed spaces and tabs                                            no-mixed-spaces-and-tabs
  38:15  error  Unexpected tab character                                         no-tabs
  38:37  error  A space is required after ','                                    comma-spacing
  39:2   error  Mixed spaces and tabs                                            no-mixed-spaces-and-tabs
  39:13  error  Unexpected tab character                                         no-tabs
  40:1   error  Expected indentation of 12 spaces but found 6 tabs               indent
  40:1   error  Unexpected tab character                                         no-tabs
42:13 error Unary operator '++' used no-plusplus
```

Comment (infograf768) [reply build/media_source/system/js/multiselect.es6.js]: http://ci.joomla.org/joomla/joomla-cms/16315/15

**Comment by Puneet-Jain-18** [comment]
@franz-wohlkoenig The PR is working as expected, what could be possible reason for the failure of continuous-integration/drone/pr test ??

**Comment by ghost** [comment]
@Puneet-Jain-18 maybe @rdeutz can help (i'm user, no Developer).

**Comment by Puneet-Jain-18** [comment]
I have tested this item :white_check_mark: successfully on fedfc5e03dc9811ba76ddb010a129851a48f2f4f<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24328">issues.joomla.org/tracker/joomla-cms/24328</a>.</sub>

**Comment by euismod2336** [comment]
I have tested this item :white_check_mark: successfully on fedfc5e03dc9811ba76ddb010a129851a48f2f4f<br /><br />works as advertised<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24328">issues.joomla.org/tracker/joomla-cms/24328</a>.</sub>

**Comment by ghost** [comment]
Status "Ready To Commit".

**Comment by roland-d** [comment]
I have tested this item :red_circle: unsuccessfully on fedfc5e03dc9811ba76ddb010a129851a48f2f4f<br /><br />I was unable to reproduce this. What I did was:
1. Check the checkbox
2. Click on the name of the article to edit it
3. Click on the browser back button
4. I get back to the list and the item is not selected anymore

This is using Firefox on Mac.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24328">issues.joomla.org/tracker/joomla-cms/24328</a>.</sub>

**Comment by roland-d** [comment]
@maliouris Thank you for the pull request unfortunately I am unable to reproduce this. Can you check on the current `4.0-dev` branch to see if you still have this issue?

**Comment by Quy** [comment]
Confirmed. OK in Firefox, but not in Chrome.

Click the checkbox next to the article.
Click on the article name to edit.
Click the Back button.

**Comment by maliouris** [comment]
I use chrome and it works for me how is that possible ? However I made that pull request way too long ago so it might not work now because some of the files that I have changed have been updated since my pull request. I don't know and at the moment I cant find out why it doesn't work as it used to do. I mean I don't have the time to do it. I will give it a try after 2 weeks

File: build/media_source/system/js/joomla-toolbar-button.w-c.es6.js [hunk 8]
```diff
@@ -44,15 +44,16 @@ window.customElements.define('joomla-toolbar-button', class extends HTMLElement
 
     this.disabled = false;
     // If list selection is required, set button to disabled by default
```
Comment (Quy) [inline build/media_source/system/js/joomla-toolbar-button.w-c.es6.js]:
```suggestion
```

**Comment by roland-d** [comment]
Removing RTC for now until issue can reliably be reproduced and tested.

**Comment by ghost** [comment]
Status back on "Pending".

**Comment by Nilesh5995** [comment]
I have tested this item :red_circle: unsuccessfully on f087257cb68da54c1e365a0490c805dfdc1150e0<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24328">issues.joomla.org/tracker/joomla-cms/24328</a>.</sub>

**Comment by nidhi1709** [comment]
No files were present while applying patch.

**Comment by ahghatol** [comment]
I have tested this item :red_circle: unsuccessfully on f087257cb68da54c1e365a0490c805dfdc1150e0<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24328">issues.joomla.org/tracker/joomla-cms/24328</a>.</sub>

**Comment by ahghatol** [comment]
I have tested this item :white_check_mark: successfully on f087257cb68da54c1e365a0490c805dfdc1150e0<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24328">issues.joomla.org/tracker/joomla-cms/24328</a>.</sub>

**Comment by deepa-g** [comment]
I have tested this item :white_check_mark: successfully on f087257cb68da54c1e365a0490c805dfdc1150e0<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24328">issues.joomla.org/tracker/joomla-cms/24328</a>.</sub>

**Comment by johndeutesfeld** [comment]
I have tested this item :red_circle: unsuccessfully on f087257cb68da54c1e365a0490c805dfdc1150e0<br /><br />1. Tested as instructed
2. Problem still occurs
Edit: using Chrome
When coming back from the edit window the buttons are still disabled. After de-selecting everything via the the top box and re-selecting the article the buttons are active again but I don't think this is the desired change.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24328">issues.joomla.org/tracker/joomla-cms/24328</a>.</sub>

**Comment by robinstern** [comment]
I have tested this item :red_circle: unsuccessfully on f087257cb68da54c1e365a0490c805dfdc1150e0<br /><br />Tested unsuccessfully!

No improvements detected!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24328">issues.joomla.org/tracker/joomla-cms/24328</a>.</sub>

**Comment by Quy** [comment]
Closing as no longer reproducible.

**Comment by maliouris** [comment]
Watch https://github.com/joomla/joomla-cms/pull/30498

**Comment by Quy** [comment]
Please test PR #30498

**Pull request closed without merging**